src/Display/Modal/Modal.tsx

```diff
@@ -13,144 +13,115 @@ import {
+      if (isVisible) onClose();
+    });
+
+    document.addEventListener('keydown', escapeKeyEventListener, false);
+
+    // This function will be called on unmount and _before_ this effect is
+    // re-executed beause its dependencies changed. This gives us the chance
+    // to clean up the existing escape key event listener.
+    return () =>
+      document.removeEventListener('keydown', escapeKeyEventListener, false);
+  }, [isVisible, onClose]);
 
   // To prevent the modal from closing
   // when a mousedown event occurs inside the ModalContentArea
   // but the subsequent mouseup event occurs outside
-  handleMouseDown = (e: React.MouseEvent) => {
-    const element = e.target as HTMLElement;
-    this.mouseDownTarget = element;
-  };
-
-  handleClick = (e: React.MouseEvent) => {
-    const element = e.target as HTMLElement;
-    if (this.mouseDownTarget === element) {
-      this.props.onClose();
-    }
-  };
-
-  render() {
-    const {
-      title,
-      onClose,
-      children,
-      className,
-      hideContentArea,
-      centering,
-      removeAnimation,
-      footer,
-      size,
-      hideHeader,
-      ...restProps
-    } = this.props;
-
-    const { isOpen } = this.state;
-
-    return (
-      <ModalContainer
-        data-testid="modal-container"
-        className={classNames('aries-modal', className)}
-        centering={centering}
-        isOpen={isOpen}
-        removeAnimation={removeAnimation}
-        onMouseDown={this.handleMouseDown}
-        onClick={this.handleClick}
-      >
-        <ModalDialog className="modal-dialog">
-          <ModalContentArea
-            className="modal-content"
-            role="dialog"
-            aria-modal="true"
-            hideContentArea={hideContentArea}
-            centering={centering}
-            onClick={e => e.stopPropagation()}
-            tabIndex={0}
-            isOpen={isOpen}
-            removeAnimation={removeAnimation}
-            size={size}
-            ref={this.modalContentAreaRef}
-            {...restProps}
-          >
-            {!hideHeader && (
-              <ModalHeader className="modal-header">
-                <h3>{title}</h3>
-                <button
-                  data-testid="close-button"
-                  type="button"
-                  onClick={onClose}
-                >
-                  <CloseIcon color={hideContentArea ? 'white' : 'grey'} />
-                </button>
-              </ModalHeader>
-            )}
-            <ModalBody className="modal-body" hideContentArea={hideContentArea}>
-              {children}
-            </ModalBody>
-            {footer !== undefined && (
-              <ModalFooter className="modal-footer">{footer}</ModalFooter>
-            )}
-          </ModalContentArea>
-        </ModalDialog>
-      </ModalContainer>
-    );
-  }
-}
+  const [mouseDownTarget, setMouseDownTarget] = React.useState<HTMLElement>(
```

Comment:
I suggest using `useRef` to keep the `mouseDownTarget`, because `useState` triggers re-render. The original implementation using `this.mouseDownTarget` to keep the value was equivalent to using `useRef` in a functional component.

Reply:
Updated!

Reply:
Updated!